internal/controller/progressive/progressive.go

```diff
@@ -303,8 +283,7 @@ func processUpgradingChild(
 				return false, false, 0, err
 			}
 
-			err = kubernetes.CreateResource(ctx, c, newUpgradingChildDef)
-			return false, true, 0, err
+			return false, newUpgradingChildDef != nil, 0, nil
```

Comment:
One thing to look at is that I ended up changing the order of marking the old "upgrading" one recyclable and creating the new "upgrading" one. Say we successfully create the new one and get a conflict error marking the previous one "recyclable". Then we will temporarily have 2 that are marked "upgrading". When we come back around to reconcile again, we will call [FindMostCurrentChildOfUpgrade()](https://github.com/numaproj/numaplane/blob/main/internal/controller/progressive/progressive.go#L103) and that will actually [mark](https://github.com/numaproj/numaplane/blob/main/internal/controller/common/rollout_controller.go#L150) the older one Recyclable.